components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -2693,7 +2693,12 @@ int shoc_init_f(Int nlev, Real *pref_mid, Int nbot_shoc, Int ntop_shoc)
   ekat::host_to_device({pref_mid}, nlev, temp_d);
   view_1d pref_mid_d(temp_d[0]);
 
+#ifndef SCREAM_MONOLITHIC_KERNELS
+  SHF::SHOCTemporaries temporaries; // we won't keep these
```

Comment:
This is a little hacky but fine IMO. The f90->cxx bridge is already slow so it doesn't matter if we reallocate temporaries for each shoc_main_f call.